File: src/55. Jump Game/55.py
from typing import List
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        max_i = 0
        for i, jump in enumerate(nums):
            if max_i >= i and i + jump > max_i:
                max_i = i + jump
        return max_i >= i
    
if __name__ == "__main__":
    nums = [2,3,1,1,4]
    print(Solution().canJump(nums))

File: src/55. Jump Game/55.java
class Solution {
    public boolean canJump(int[] nums) {
        int max_i = 0, n = nums.length ;
        for (int i = 0; i < n; i ++) {
            if (max_i >= i && i + nums[i] > max_i) {
                max_i = i + nums[i];
            }
        }
        return max_i >= n - 1;
    }
}